### ISSUE TITLE
FilterFormInput always has allowEmpty=true props

### ISSUE THREAD

**Issue by ivan24** [marmelab/react-admin#4499]
**What you were expecting:**
FilterFormInput should take prop allowEmpty from Input component but allowEmpty prop is ignore and always set true. See https://github.com/marmelab/react-admin/blob/v3.3.0/packages/ra-ui-materialui/src/list/FilterFormInput.js#L46

**What happened instead:**
FilterFormInput has always allowEmpty=true despite of input props

**Steps to reproduce:**
Set allowEmpty={false} for Input component
```
  <Filter {...props}>
    <SearchInput source="q" />
    <ReferenceInput
      source="post_id"
      reference="posts"
      allowEmpty={false}
      alwaysOn
    >
      <SelectInput optionText="title" />
    </ReferenceInput>
  </Filter>
```

**Related code:**
CodeSandbox: https://codesandbox.io/s/ecstatic-sea-remus/
(https://remus.csb.app/#/comments)

**Environment**

* React-admin version:
3.3.0
* Last version that did not exhibit the issue (if applicable):
2.9.0
* React version:
16.12
* Browser:
chrome

**Comment by JulienMattiussi** [comment]
Not sure why this prop have been forced true.
But I think it should be opened.
Thanks for your return and the sandbox.
I'm flagging this as Enhancement.

**Comment by fzaninotto** [comment]
Forcing `allowEmpty` to true for forms is necessary to allow a user to disable a SelectInput filter. It should remain the default.

**Comment by fzaninotto** [comment]
Fixed by #4500
